File: code/ep2p/ep2p-core/src/main/java/com/yscf/core/model/ptp/financial/BizWithdrawModel.java
/*
 **************************************************************************
 * 版权声明：
 * 本软件为深圳市力泰金融信息服务有限公司开发研制。未经本公司正式书面同意，
 * 其他任何个人、团体不得使用、复制、修改或发布本软件. 
 **************************************************************************
 * 程序描述:
 * TODO
 * 
 **************************************************************************
 * 修改历史:  
 * Date:       	    by:    		    Reason:  
 *           
 * 2015年11月3日     JunJie.Liu		Initial Version.
 *************************************************************************
 */
package com.yscf.core.model.ptp.financial;

import java.math.BigDecimal;


/**
 * Description：<br>
 * 提现VO
 * 
 * @author JunJie.Liu
 * @date 2015年11月3日
 * @version v1.0.0
 */
public class BizWithdrawModel {

	private String customerName;

	private String sname;

	private String phoneNo;

	private String applyTime;// 申请时间

	private BigDecimal amount;// 提现金额

	private BigDecimal fee;// 手续费

	private String belongingBank;// 银行

	private String openAddress;// 银行支行

	private String bankcardNo; // 卡号

	private String auditStatus; // 状态

	private String auditStatusName;// 状态名称

	private String witDesc;// 备注

	private String witSureDesc;// 用户名

	private String queryStatus; // 类型

	public String getCustomerName() {
		return customerName;
	}

	public void setCustomerName(String customerName) {
		this.customerName = customerName;
	}

	public String getSname() {
		return sname;
	}

	public void setSname(String sname) {
		this.sname = sname;
	}

	public String getPhoneNo() {
		return phoneNo;
	}

	public void setPhoneNo(String phoneNo) {
		this.phoneNo = phoneNo;
	}

	public String getApplyTime() {
		return applyTime;
	}

	public void setApplyTime(String applyTime) {
		this.applyTime = applyTime;
	}

	public BigDecimal getAmount() {
		return amount;
	}

	public void setAmount(BigDecimal amount) {
		this.amount = amount;
	}

	public BigDecimal getFee() {
		return fee;
	}

	public void setFee(BigDecimal fee) {
		this.fee = fee;
	}

	public String getBelongingBank() {
		return belongingBank;
	}

	public void setBelongingBank(String belongingBank) {
		this.belongingBank = belongingBank;
	}

	public String getOpenAddress() {
		return openAddress;
	}

	public void setOpenAddress(String openAddress) {
		this.openAddress = openAddress;
	}

	public String getBankcardNo() {
		return bankcardNo;
	}

	public void setBankcardNo(String bankcardNo) {
		this.bankcardNo = bankcardNo;
	}

	public String getAuditStatus() {
		return auditStatus;
	}

	public void setAuditStatus(String auditStatus) {
		this.auditStatus = auditStatus;
	}

	public String getAuditStatusName() {
		return auditStatusName;
	}

	public void setAuditStatusName(String auditStatusName) {
		this.auditStatusName = auditStatusName;
	}

	public String getWitDesc() {
		return witDesc;
	}

	public void setWitDesc(String witDesc) {
		this.witDesc = witDesc;
	}

	public String getWitSureDesc() {
		return witSureDesc;
	}

	public void setWitSureDesc(String witSureDesc) {
		this.witSureDesc = witSureDesc;
	}

	/**
	 * 获取状态
	 * 
	 * @param value
	 * @param row
	 * @param index
	 * @returns {String}
	 */
	public String formatAuditStatus(String auditStatus) {
		// 审核状态（1：申请中、2：同意（确认中）、3：拒绝、4：转账成功、5：转账失败）
		if ("1".equals(auditStatus)) {
			return "提现申请";
		} else if ("2".equals(auditStatus)) {
			return "转账确认";
		} else if ("3".equals(auditStatus)) {
			return "提现拒绝";
		} else if ("4".equals(auditStatus)) {
			return "转账成功";
		} else if ("5".equals(auditStatus)) {
			return "转账失败";
		}
		return "";
	}

	public String getQueryStatus() {
		return queryStatus;
	}

	public void setQueryStatus(String queryStatus) {
		this.queryStatus = queryStatus;
	}
	
	
}

File: code/ep2p/ep2p-system/WebRoot/resources/js/finance/model/biz_withdraw_model.js
/**
 *业务提现审核类js_model
 */
var biz_withdraw_model = [[    
       	        {field:'pid',checkbox:true},
	       	    {field:'customerName',title:'客户名称',width:'10%',align:'center',formatter:common.getCustomerName},    
	            {field:'sname',title:'真实姓名',width:'10%',align:'center',formatter:common.getSName}, 
	            {field:'phoneNo',title:'手机号码',width:'10%',align:'center',formatter:common.getPhoneNo},
    	        
    	        
    	        {field:'applyTime',title:'申请时间',width:'12%',align:'center',sortable:true},    
    	        {field:'amount',title:'提现金额',width:'8%',align:'center',sortable:true,formatter:common.formatCurrency}, 
    	        {field:'fee',title:'提现手续费',width:'8%',align:'center',sortable:true,formatter:common.formatCurrency},
    	        {field:'belongingBank',title:'所属银行',width:'10%',align:'center',sortable:true,formatter:belongingBank},
    	        {field:'openAddress',title:'开户支行',width:'10%',align:'center',sortable:true,formatter:openAddress},
    	        {field:'bankcardNo',title:'银行卡号',width:'10%',align:'center',sortable:true,formatter:bankcardNo},
    	        {field:'auditStatus',title:'状态',width:'6%',align:'center',sortable:true,formatter:formatAuditStatus},
    	        {field:'userName',title:'处理人',width:'6%',align:'center',sortable:true,formatter:getUserName},
    	        {field:'witDesc',title:'备注',width:'20%',align:'center',sortable:true,formatter:getWitDesc}
    	    ]];

/**
 * 所属银行取值
 * @param value
 * @param row
 * @param index
 * @returns
 */
function belongingBank(value,row,index){
	if(row.bank){
		return row.bank.belongingBank;
	}
	return "";
}
/**
 * 开户支行取值
 * @param value
 * @param row
 * @param index
 * @returns
 */
function openAddress(value,row,index){
	if(row.bank){
		return row.bank.openAddress;
	}
	return "";
}
/**
 * 银行卡号取值
 * @param value
 * @param row
 * @param index
 * @returns
 */
function bankcardNo(value,row,index){
	if(row.bank){
		return row.bank.bankcardNo;
	}
	return "";
}

/**
 * 获取备注
 * @param value
 * @param row
 * @param index
 */
function getWitDesc(value,row,index){
	if(row.witSureDesc){
		return row.witSureDesc;
	}
	return value;
}

/**
 * 获取用户名
 * @param value
 * @param row
 * @param index
 * @returns
 */
function getUserName(value,row,index){
	if(row.transferedUserName){
		return row.transferedUserName;
	}
	return value;
}

/**
 * 获取状态
 * @param value
 * @param row
 * @param index
 * @returns {String}
 */
function formatAuditStatus(value,row,index){
	//审核状态（1：申请中、2：同意（确认中）、3：拒绝、4：转账成功、5：转账失败）
	if(row.auditStatus == 1){
		return "提现审核";
	}else if(row.auditStatus == 2){
		return "转账确认";
	}else if(row.auditStatus == 3){
		return "提现拒绝";
	}else if(row.auditStatus == 4){
		return "转账成功";
	}else if(row.auditStatus == 5){
		return "转账失败";
	}
	return "";
}

